mmv1/products/discoveryengine/TargetSite.yaml

```diff
@@ -0,0 +1,184 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'TargetSite'
+description: |
+  TargetSite represents a URI pattern that the users want to confine their
+  search.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.dataStores.siteSearchEngine.targetSites'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/siteSearchEngine/targetSites'
+self_link: '{{name}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/siteSearchEngine/targetSites'
+update_verb: :PATCH
+update_mask: false
+delete_url: '{{name}}'
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/default_collection/dataStores/{{data_store_id}}/siteSearchEngine/targetSites/{{target_site_id}}',
+  ]
+
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_targetsite_basic"
+    primary_resource_id: 'basic'
+    vars:
+      data_store_id: "data-store-id"
+    ignore_read_extra:
+      - 'project'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  post_create: templates/terraform/post_create/set_computed_name.erb
+  post_update: templates/terraform/post_update/set_computed_name.erb
```

Comment:
I don't think we need this post_update as well. It should work only after create because the name should never change

Does it work if you remove this?

Reply:
Hi Sam, I tested locally, after removing this `post_update`, the update test couldn't pass because the resource name changed.

I ran cli `TF_LOG=DEBUG make testacc TEST=./google/services/discoveryengine TESTARGS='-run=TestAccDiscoveryEngineTargetSite_discoveryengineTargetsiteBasicExample_update' > output.log` to dump out debugging log, and got results: https://screenshot.googleplex.com/4izdznRcKyJGs9j

Reply:
Double confirmed with the API owner, we plan not to support udpate operation because:
1. Update API is not used by UI (Pantheon);
2. Update API has not been fully tested, and now we expect the customer to do delete-then-recreate for an update.

Thus, for this newly supported resource `TargetSite`, we will support 4 APIs: get, list, create, delete
I will do the following things: 
1. remove this `post_update` custom code and update test
2. mark this resource as immutable